### PR TITLE
Fix two logical errors when reading from stdin in clickhouse local

### LINKED ISSUE #71043: 'Unknown executor' when reading from stdin in clickhouse local
```
% ~/ClickHouse/build/programs/clickhouse local -q "create table a (x Int64) engine Memory; insert into a select * from input('auto') format Parquet" < t.parquet 
2024.10.25 04:02:17.781909 [ 1947317 ] {ae77349d-5b46-4bb5-a3e9-fb26ccf3891d} <Fatal> : Logical error: 'Unknown executor'.
2024.10.25 04:02:17.937971 [ 1947317 ] {ae77349d-5b46-4bb5-a3e9-fb26ccf3891d} <Fatal> : Stack trace (when copying this message, always include the lines below):

0. /home/ubuntu/ClickHouse/contrib/llvm-project/libcxx/include/exception:141: Poco::Exception::Exception(String const&, int) @ 0x00000000169df5b2
1. /home/ubuntu/ClickHouse/src/Common/Exception.cpp:109: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c490f19
2. /home/ubuntu/ClickHouse/src/Common/Exception.h:110: DB::Exception::Exception(PreformattedMessage&&, int) @ 0x000000000716920c
3. /home/ubuntu/ClickHouse/src/Common/Exception.h:128: DB::Exception::Exception<>(int, FormatStringHelperImpl<>) @ 0x000000000716c0cb
4. /home/ubuntu/ClickHouse/src/Client/LocalConnection.cpp:325: DB::LocalConnection::sendData(DB::Block const&, String const&, bool) @ 0x000000001300f6f0
5. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1823: DB::ClientBase::sendDataFromPipe(DB::Pipe&&, std::shared_ptr<DB::IAST>, bool) @ 0x0000000012fc25f6
6. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1779: DB::ClientBase::sendDataFrom(DB::ReadBuffer&, DB::Block&, DB::ColumnsDescription const&, std::shared_ptr<DB::IAST>, bool) @ 0x0000000012fc2c00
7. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1843: DB::ClientBase::sendDataFromStdin(DB::Block&, DB::ColumnsDescription const&, std::shared_ptr<DB::IAST>) @ 0x0000000012fc2978
8. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1750: DB::ClientBase::sendData(DB::Block&, DB::ColumnsDescription const&, std::shared_ptr<DB::IAST>) @ 0x0000000012fc172d
9. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1610: DB::ClientBase::processInsertQuery(String const&, std::shared_ptr<DB::IAST>) @ 0x0000000012fc04d9
10. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2075: DB::ClientBase::processParsedSingleQuery(String const&, String const&, std::shared_ptr<DB::IAST>, std::optional<bool>, bool) @ 0x0000000012fbaa1a
11. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2411: DB::ClientBase::executeMultiQuery(String const&) @ 0x0000000012fc3ac3
12. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2556: DB::ClientBase::processQueryText(String const&) @ 0x0000000012fc47a6
13. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2860: DB::ClientBase::runNonInteractive() @ 0x0000000012fc6f1b
14. /home/ubuntu/ClickHouse/programs/local/LocalServer.cpp:605: DB::LocalServer::main(std::vector<String, std::allocator<String>> const&) @ 0x000000000c6a5906
15. /home/ubuntu/ClickHouse/base/poco/Util/src/Application.cpp:315: Poco::Util::Application::run() @ 0x0000000016aa68e6
16. /home/ubuntu/ClickHouse/programs/local/LocalServer.cpp:1026: mainEntryClickHouseLocal(int, char**) @ 0x000000000c6af0d5
17. /home/ubuntu/ClickHouse/programs/main.cpp:248: main @ 0x0000000007162c6b
18. ? @ 0x00007ffff7c29d90
19. ? @ 0x00007ffff7c29e40
20. _start @ 0x000000000716202e

2024.10.25 04:02:17.938600 [ 1947327 ] {} <Fatal> ClientBase: ########## Short fault info ############
2024.10.25 04:02:17.938646 [ 1947327 ] {} <Fatal> ClientBase: (version 24.10.1.1, build id: , git hash: 9e2ae7e0c7ec9eb73143141dda372eda736f391d, architecture: x86_64) (from thread 1947317) Received signal 6
2024.10.25 04:02:17.938664 [ 1947327 ] {} <Fatal> ClientBase: Signal description: Aborted
2024.10.25 04:02:17.938672 [ 1947327 ] {} <Fatal> ClientBase: 
2024.10.25 04:02:17.938689 [ 1947327 ] {} <Fatal> ClientBase: Stack trace: 0x000000000c4bf488 0x000000000c70bc45 0x00007ffff7c42520 0x00007ffff7c969fd 0x00007ffff7c42476 0x00007ffff7c287f3 0x000000000c490613 0x000000000c4910d4 0x000000000716920c 0x000000000716c0cb 0x000000001300f6f0 0x0000000012fc25f6 0x0000000012fc2c00 0x0000000012fc2978 0x0000000012fc172d 0x0000000012fc04d9 0x0000000012fbaa1a 0x0000000012fc3ac3 0x0000000012fc47a6 0x0000000012fc6f1b 0x000000000c6a5906 0x0000000016aa68e6 0x000000000c6af0d5 0x0000000007162c6b 0x00007ffff7c29d90 0x00007ffff7c29e40 0x000000000716202e
2024.10.25 04:02:17.938700 [ 1947327 ] {} <Fatal> ClientBase: ########################################
2024.10.25 04:02:17.938714 [ 1947327 ] {} <Fatal> ClientBase: (version 24.10.1.1, build id: , git hash: 9e2ae7e0c7ec9eb73143141dda372eda736f391d) (from thread 1947317) (query_id: ae77349d-5b46-4bb5-a3e9-fb26ccf3891d) (query: insert into a select * from input('auto') format Parquet) Received signal Aborted (6)
2024.10.25 04:02:17.938724 [ 1947327 ] {} <Fatal> ClientBase: 
2024.10.25 04:02:17.938731 [ 1947327 ] {} <Fatal> ClientBase: Stack trace: 0x000000000c4bf488 0x000000000c70bc45 0x00007ffff7c42520 0x00007ffff7c969fd 0x00007ffff7c42476 0x00007ffff7c287f3 0x000000000c490613 0x000000000c4910d4 0x000000000716920c 0x000000000716c0cb 0x000000001300f6f0 0x0000000012fc25f6 0x0000000012fc2c00 0x0000000012fc2978 0x0000000012fc172d 0x0000000012fc04d9 0x0000000012fbaa1a 0x0000000012fc3ac3 0x0000000012fc47a6 0x0000000012fc6f1b 0x000000000c6a5906 0x0000000016aa68e6 0x000000000c6af0d5 0x0000000007162c6b 0x00007ffff7c29d90 0x00007ffff7c29e40 0x000000000716202e
2024.10.25 04:02:17.954077 [ 1947327 ] {} <Fatal> ClientBase: 0.0. inlined from /home/ubuntu/ClickHouse/src/Common/StackTrace.cpp:380: StackTrace::tryCapture()
2024.10.25 04:02:17.954097 [ 1947327 ] {} <Fatal> ClientBase: 0. /home/ubuntu/ClickHouse/src/Common/StackTrace.cpp:349: StackTrace::StackTrace(ucontext_t const&) @ 0x000000000c4bf488
2024.10.25 04:02:17.970569 [ 1947327 ] {} <Fatal> ClientBase: 1. /home/ubuntu/ClickHouse/src/Common/SignalHandlers.cpp:85: signalHandler(int, siginfo_t*, void*) @ 0x000000000c70bc45
2024.10.25 04:02:17.970586 [ 1947327 ] {} <Fatal> ClientBase: 2. ? @ 0x00007ffff7c42520
2024.10.25 04:02:17.970594 [ 1947327 ] {} <Fatal> ClientBase: 3. ? @ 0x00007ffff7c969fd
2024.10.25 04:02:17.970601 [ 1947327 ] {} <Fatal> ClientBase: 4. ? @ 0x00007ffff7c42476
2024.10.25 04:02:17.970609 [ 1947327 ] {} <Fatal> ClientBase: 5. ? @ 0x00007ffff7c287f3
2024.10.25 04:02:17.999727 [ 1947327 ] {} <Fatal> ClientBase: 6. /home/ubuntu/ClickHouse/src/Common/Exception.cpp:48: DB::abortOnFailedAssertion(String const&, void* const*, unsigned long, unsigned long) @ 0x000000000c490613
2024.10.25 04:02:18.026712 [ 1947327 ] {} <Fatal> ClientBase: 7.0. inlined from /home/ubuntu/ClickHouse/src/Common/Exception.cpp:71: DB::handle_error_code(String const&, int, bool, std::vector<void*, std::allocator<void*>> const&)
2024.10.25 04:02:18.026734 [ 1947327 ] {} <Fatal> ClientBase: 7. /home/ubuntu/ClickHouse/src/Common/Exception.cpp:115: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c4910d4
2024.10.25 04:02:18.040726 [ 1947327 ] {} <Fatal> ClientBase: 8.0. inlined from /home/ubuntu/ClickHouse/src/Common/Exception.h:110: Exception
2024.10.25 04:02:18.040753 [ 1947327 ] {} <Fatal> ClientBase: 8. /home/ubuntu/ClickHouse/src/Common/Exception.h:63: DB::Exception::Exception(PreformattedMessage&&, int) @ 0x000000000716920c
2024.10.25 04:02:18.056422 [ 1947327 ] {} <Fatal> ClientBase: 9. /home/ubuntu/ClickHouse/src/Common/Exception.h:128: DB::Exception::Exception<>(int, FormatStringHelperImpl<>) @ 0x000000000716c0cb
2024.10.25 04:02:18.092829 [ 1947327 ] {} <Fatal> ClientBase: 10. /home/ubuntu/ClickHouse/src/Client/LocalConnection.cpp:325: DB::LocalConnection::sendData(DB::Block const&, String const&, bool) @ 0x000000001300f6f0
2024.10.25 04:02:18.174353 [ 1947327 ] {} <Fatal> ClientBase: 11. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1823: DB::ClientBase::sendDataFromPipe(DB::Pipe&&, std::shared_ptr<DB::IAST>, bool) @ 0x0000000012fc25f6
2024.10.25 04:02:18.251906 [ 1947327 ] {} <Fatal> ClientBase: 12. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1779: DB::ClientBase::sendDataFrom(DB::ReadBuffer&, DB::Block&, DB::ColumnsDescription const&, std::shared_ptr<DB::IAST>, bool) @ 0x0000000012fc2c00
2024.10.25 04:02:18.352710 [ 1947327 ] {} <Fatal> ClientBase: 13. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1843: DB::ClientBase::sendDataFromStdin(DB::Block&, DB::ColumnsDescription const&, std::shared_ptr<DB::IAST>) @ 0x0000000012fc2978
2024.10.25 04:02:18.452484 [ 1947327 ] {} <Fatal> ClientBase: 14. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1750: DB::ClientBase::sendData(DB::Block&, DB::ColumnsDescription const&, std::shared_ptr<DB::IAST>) @ 0x0000000012fc172d
2024.10.25 04:02:18.530158 [ 1947327 ] {} <Fatal> ClientBase: 15. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1610: DB::ClientBase::processInsertQuery(String const&, std::shared_ptr<DB::IAST>) @ 0x0000000012fc04d9
2024.10.25 04:02:18.609927 [ 1947327 ] {} <Fatal> ClientBase: 16. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2075: DB::ClientBase::processParsedSingleQuery(String const&, String const&, std::shared_ptr<DB::IAST>, std::optional<bool>, bool) @ 0x0000000012fbaa1a
2024.10.25 04:02:18.712688 [ 1947327 ] {} <Fatal> ClientBase: 17. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2411: DB::ClientBase::executeMultiQuery(String const&) @ 0x0000000012fc3ac3
2024.10.25 04:02:18.816316 [ 1947327 ] {} <Fatal> ClientBase: 18. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2556: DB::ClientBase::processQueryText(String const&) @ 0x0000000012fc47a6
2024.10.25 04:02:18.922996 [ 1947327 ] {} <Fatal> ClientBase: 19. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2860: DB::ClientBase::runNonInteractive() @ 0x0000000012fc6f1b
2024.10.25 04:02:18.944515 [ 1947327 ] {} <Fatal> ClientBase: 20. /home/ubuntu/ClickHouse/programs/local/LocalServer.cpp:605: DB::LocalServer::main(std::vector<String, std::allocator<String>> const&) @ 0x000000000c6a5906
2024.10.25 04:02:18.954627 [ 1947327 ] {} <Fatal> ClientBase: 21. /home/ubuntu/ClickHouse/base/poco/Util/src/Application.cpp:315: Poco::Util::Application::run() @ 0x0000000016aa68e6
2024.10.25 04:02:19.002679 [ 1947327 ] {} <Fatal> ClientBase: 22. /home/ubuntu/ClickHouse/programs/local/LocalServer.cpp:1026: mainEntryClickHouseLocal(int, char**) @ 0x000000000c6af0d5
2024.10.25 04:02:19.004995 [ 1947327 ] {} <Fatal> ClientBase: 23. /home/ubuntu/ClickHouse/programs/main.cpp:248: main @ 0x0000000007162c6b
2024.10.25 04:02:19.005009 [ 1947327 ] {} <Fatal> ClientBase: 24. ? @ 0x00007ffff7c29d90
2024.10.25 04:02:19.005017 [ 1947327 ] {} <Fatal> ClientBase: 25. ? @ 0x00007ffff7c29e40
2024.10.25 04:02:19.100595 [ 1947327 ] {} <Fatal> ClientBase: 26. _start @ 0x000000000716202e
2024.10.25 04:02:19.100619 [ 1947327 ] {} <Fatal> ClientBase: This ClickHouse version is not official and should be upgraded to the official build.
2024.10.25 04:02:19.100729 [ 1947327 ] {} <Fatal> ClientBase: Changed settings: allow_introspection_functions = true, storage_file_read_method = 'mmap'
zsh: IOT instruction (core dumped)  ~/ClickHouse/build/programs/clickhouse local -q  < t.parquet
```

But works if you `cat t.parquet | clickhouse local ...` instead of `clickhouse local ... < t.parquet`

### LINKED ISSUE #71044: 'Input initializer is not set' when reading from stdin in clickhouse local
```sql
% echo 'meow' | ~/ClickHouse/build/programs/clickhouse local -q "insert into function file('a.csv') select * from input('x Int64') format CSV"                                                                                    
2024.10.25 05:30:15.521604 [ 1950607 ] {c7f6c1e4-c6f1-4d52-9ed8-cd10a85200aa} <Fatal> : Logical error: 'Input initializer is not set'.                                                                                                                                                
2024.10.25 05:30:15.682231 [ 1950607 ] {c7f6c1e4-c6f1-4d52-9ed8-cd10a85200aa} <Fatal> : Stack trace (when copying this message, always include the lines below):                                                                                                                      
                                                                                                                                                                                                                                                                                      
0. /home/ubuntu/ClickHouse/contrib/llvm-project/libcxx/include/exception:141: Poco::Exception::Exception(String const&, int) @ 0x00000000169df5b2                                                                                                                                     
1. /home/ubuntu/ClickHouse/src/Common/Exception.cpp:109: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c490f19                                                                                                                                     
2. /home/ubuntu/ClickHouse/src/Common/Exception.h:110: DB::Exception::Exception(PreformattedMessage&&, int) @ 0x000000000716920c                                                                                                                                                      
3. /home/ubuntu/ClickHouse/src/Common/Exception.h:128: DB::Exception::Exception<>(int, FormatStringHelperImpl<>) @ 0x000000000716c0cb                                                                                                                                                 
4. /home/ubuntu/ClickHouse/src/Interpreters/Context.cpp:5069: DB::Context::initializeInput(std::shared_ptr<DB::IStorage> const&) @ 0x00000000112eef6e                                                                                                                                 
5. /home/ubuntu/ClickHouse/src/Storages/StorageInput.cpp:97: DB::StorageInput::read(DB::QueryPlan&, std::vector<String, std::allocator<String>> const&, std::shared_ptr<DB::StorageSnapshot> const&, DB::SelectQueryInfo&, std::shared_ptr<DB::Context const>, DB::QueryProcessingStag
e::Enum, unsigned long, unsigned long) @ 0x0000000012650054                                                                                                                                                                                                                           
6. /home/ubuntu/ClickHouse/src/Planner/PlannerJoinTree.cpp:921: DB::(anonymous namespace)::buildQueryPlanForTableExpression(std::shared_ptr<DB::IQueryTreeNode>, DB::SelectQueryInfo const&, DB::SelectQueryOptions const&, std::shared_ptr<DB::PlannerContext>&, bool, bool) @ 0x0000
000011a221ec                                                                                                                                                                                                                                                                          
7. /home/ubuntu/ClickHouse/src/Planner/PlannerJoinTree.cpp:1835: DB::buildJoinTreeQueryPlan(std::shared_ptr<DB::IQueryTreeNode> const&, DB::SelectQueryInfo const&, DB::SelectQueryOptions&, std::unordered_set<String, std::hash<String>, std::equal_to<String>, std::allocator<Strin
g>> const&, std::shared_ptr<DB::PlannerContext>&) @ 0x0000000011a1b7e2                                                                                                                                                                                                                
8. /home/ubuntu/ClickHouse/src/Planner/Planner.cpp:1535: DB::Planner::buildPlanForQueryNode() @ 0x0000000011a06eae                                                                                                                                                                    
9. /home/ubuntu/ClickHouse/src/Planner/Planner.cpp:1289: DB::Planner::buildQueryPlanIfNeeded() @ 0x0000000011a03c7e                                                                                                                                                                   
10. /home/ubuntu/ClickHouse/src/Interpreters/InterpreterSelectQueryAnalyzer.cpp:253: DB::InterpreterSelectQueryAnalyzer::buildQueryPipeline() @ 0x0000000011a01c79                                                                                                                    
11. /home/ubuntu/ClickHouse/src/Interpreters/InterpreterInsertQuery.cpp:519: DB::InterpreterInsertQuery::buildInsertSelectPipeline(DB::ASTInsertQuery&, std::shared_ptr<DB::IStorage>) @ 0x00000000119a0669                                                                           
12. /home/ubuntu/ClickHouse/src/Interpreters/InterpreterInsertQuery.cpp:810: DB::InterpreterInsertQuery::execute() @ 0x00000000119a400c                                                                                                                                               
13. /home/ubuntu/ClickHouse/src/Interpreters/executeQuery.cpp:1346: DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x0000000011ccae96                                                
14. /home/ubuntu/ClickHouse/src/Interpreters/executeQuery.cpp:1513: DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x0000000011cc6c7d                                                                                
15. /home/ubuntu/ClickHouse/src/Client/LocalConnection.cpp:240: DB::LocalConnection::sendQuery(DB::ConnectionTimeouts const&, String const&, std::unordered_map<String, String, std::hash<String>, std::equal_to<String>, std::allocator<std::pair<String const, String>>> const&, Str
ing const&, unsigned long, DB::Settings const*, DB::ClientInfo const*, bool, std::function<void (DB::Progress const&)>) @ 0x000000001300e4ea                                                                                                                                          
16. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:1587: DB::ClientBase::processInsertQuery(String const&, std::shared_ptr<DB::IAST>) @ 0x0000000012fc033e                                                                                                                         
17. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2075: DB::ClientBase::processParsedSingleQuery(String const&, String const&, std::shared_ptr<DB::IAST>, std::optional<bool>, bool) @ 0x0000000012fbaa1a                                                                         
18. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2411: DB::ClientBase::executeMultiQuery(String const&) @ 0x0000000012fc3ac3                                                                                                                                                     
19. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2556: DB::ClientBase::processQueryText(String const&) @ 0x0000000012fc47a6                                                                                                                                                      
20. /home/ubuntu/ClickHouse/src/Client/ClientBase.cpp:2860: DB::ClientBase::runNonInteractive() @ 0x0000000012fc6f1b                                                                                                                                                                  
21. /home/ubuntu/ClickHouse/programs/local/LocalServer.cpp:605: DB::LocalServer::main(std::vector<String, std::allocator<String>> const&) @ 0x000000000c6a5906                                                                                                                        
22. /home/ubuntu/ClickHouse/base/poco/Util/src/Application.cpp:315: Poco::Util::Application::run() @ 0x0000000016aa68e6                                                                                                                                                               
23. /home/ubuntu/ClickHouse/programs/local/LocalServer.cpp:1026: mainEntryClickHouseLocal(int, char**) @ 0x000000000c6af0d5                                                                                                                                                           
24. /home/ubuntu/ClickHouse/programs/main.cpp:248: main @ 0x0000000007162c6b                                                                                                                                                                                                          
25. ? @ 0x00007ffff7c29d90                                                                                                                                                                                                                                                            
26. ? @ 0x00007ffff7c29e40                                                                                                                                                                                                                                                            
27. _start @ 0x000000000716202e
```

### PR DESCRIPTION
**Pull request by al13n321**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Closes https://github.com/ClickHouse/ClickHouse/issues/71043
Closes https://github.com/ClickHouse/ClickHouse/issues/71044

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit e19bf218f69448c9605f269ae7a3894bc24f0003 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71046/e19bf218f69448c9605f269ae7a3894bc24f0003/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/71037/b6c959846de31f489b92613cfb72d8bccfd85118/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: programs/local/LocalServer.cpp [hunk 1]
```diff
@@ -130,6 +130,11 @@ void applySettingsOverridesForLocal(ContextMutablePtr context)
     context->setSettings(settings);
 }
 
+LocalServer::LocalServer()
+{
+    is_local = true;
+}
+
```
Comment (tuanpach) [inline programs/local/LocalServer.cpp:137]:
```suggestion
LocalServer::LocalServer() : is_local(true)
{
}

```

File: src/Client/ClientBase.cpp [hunk 2]
```diff
@@ -1748,7 +1748,8 @@ void ClientBase::sendData(Block & sample, const ColumnsDescription & columns_des
     }
     else if (!is_interactive)
     {
-        sendDataFromStdin(sample, columns_description_for_query, parsed_query);
+        if (!is_local)
+            sendDataFromStdin(sample, columns_description_for_query, parsed_query);
```
Comment (tuanpach) [inline src/Client/ClientBase.cpp:1752]: Could you explain why this helps to solve the issue?

File: programs/local/LocalServer.cpp (hunk 1 above)
Comment (tuanpach) [reply programs/local/LocalServer.cpp:137]: My bad, I thought it was in the derived class.

File: src/Client/ClientBase.cpp (hunk 2 above)
Comment (tuanpach) [reply src/Client/ClientBase.cpp:1752]: This logic handles the specific logic of the derived class in the base class.

I think we can create a virtual function to separate the logic of the base and derived class:

```
// in ClientBase
virtual void sendDataWithNonInteractiveInterface(...) {
    sendDataFromStdin(sample, columns_description_for_query, parsed_query);
}

// in LocalServer
void sendDataWithNonInteractiveInterface(...) override {
    // Do nothing
}
```

In `else if (!is_interactive)` block, we can just call `sendDataWithNonInteractiveInterface(...)`. And we don't need to introduce `is_local` in `ClientBase`.

**Comment by al13n321** [comment]
(That change broke regular inserts. Here's a correct fix. And a bonus fix for a mostly-unrelated bug, because merging two PRs that touch the same code is a pain.)

File: src/Interpreters/InterpreterInsertQuery.cpp [hunk 3]
```diff
@@ -121,8 +121,7 @@ StoragePtr InterpreterInsertQuery::getTable(ASTInsertQuery & query)
 
             if (current_context->getSettingsRef()[Setting::allow_experimental_analyzer])
             {
-                InterpreterSelectQueryAnalyzer interpreter_select(query.select, current_context, select_query_options);
-                header_block = interpreter_select.getSampleBlock();
+                header_block = InterpreterSelectQueryAnalyzer::getSampleBlock(query.select, current_context, select_query_options);
```
Comment (al13n321) [inline src/Interpreters/InterpreterInsertQuery.cpp:124]: The difference here is that the static `getSampleBlock` sets `only_analyze = true` in SelectQueryOptions, which prevents query analysis from calling `IStorage::read` on the tables (which breaks `input(...)` table function in particular).

**Comment by tuanpach** [comment]
It LGTM. However, I do not seem to have permission to approve it. I can't find the `Approve` button.

**Comment by al13n321** [comment]
https://github.com/ClickHouse/ClickHouse/pull/71046/files -> "Review changes" (top right) -> "Approve"

**Review by tuanpach — APPROVED** [review]

**Pull request merged**